### PR TITLE
Migrate Orca Net to BigDL 2.0

### PR DESCRIPTION
**Pull request by charlieJ107**
Migrate orca.net to bigdl-2.0

## Changes
* Replace import from `zoo`  to `bigdl.dllib`
* Update Licenses

## Related Issue
https://github.com/intel-analytics/analytics-zoo/issues/4804

### CODE REVIEW

**Comment by jason-dai** [comment]
Some of the classes are already in `dllib`: https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net? @Le-Zheng @yangw1234

**Comment by Le-Zheng** [comment]
> Some of the classes are already in `dllib`: https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net? @Le-Zheng @yangw1234

Yes. net/utils.py is already migrated.  Need to migrate net/ other classes.

**Comment by Le-Zheng** [comment]
@charlieJ107 I find https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.

**Comment by yangw1234** [comment]
> @charlieJ107 I find https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.

Can we do this in another pr?

File: python/orca/src/bigdl/orca/net/graph_net.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+
+from bigdl.nn.layer import Model as BModel
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/net/graph_net.py:19]: need to change to `bigdl.dllib.nn.layer`

File: python/orca/src/bigdl/orca/net/graph_net.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+from bigdl.dllib.feature.image import ImageSet
+from bigdl.dllib.feature.text import TextSet
+from bigdl.dllib.keras.base import ZooKerasLayer
+from bigdl.dllib.keras.utils import *
+from bigdl.nn.layer import Layer
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/net/graph_net.py:24]: need to change to `bigdl.dllib.nn.layer`

File: python/orca/src/bigdl/orca/net/graph_net.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+from bigdl.dllib.utils import callZooFunc
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/net/graph_net.py:25]: `from bigdl.dllib.utils.file_utils import callZooFunc`

**Comment by charlieJ107** [comment]
> > @charlieJ107 I find https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> 
> Can we do this in another pr?

Seems these two files are same. Which one should we delete?

File: python/orca/src/bigdl/orca/net/net_load.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import importlib
+import os
+import sys
+
+from bigdl.dllib.utils import callZooFunc
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/net/net_load.py:21]: `from bigdl.dllib.utils.file_utils import callZooFunc`

File: python/orca/src/bigdl/orca/net/net_load.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+from bigdl.nn.layer import Model as BModel
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/net/net_load.py:22]: `bigdl.dllib.nn.layer`

File: python/orca/src/bigdl/orca/net/utils.py [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+                                      all_required_inputs]
+        raise ValueError("You should use all the placeholders that are defined in dataset, " +
+                         "%s are not used" % inputs_not_require_by_loss)
+
+
+def to_bigdl_optim_method(koptim_method):
+    # koptim_method is always an object
+    import tensorflow.keras.backend as K
+    import tensorflow.keras.optimizers as koptimizers
+    import bigdl.optim.optimizer as boptimizer
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/net/utils.py:93]: `import bigdl.dllib.optim.optimizer`

Comment (charlieJ107) [reply python/orca/src/bigdl/orca/net/utils.py]: Fixed in https://github.com/intel-analytics/analytics-zoo/pull/4830/commits/c3712bbe0c0ce9f65c7964f6f74c01d11146c0fb

**Comment by Le-Zheng** [comment]
> > @charlieJ107 I find https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> 
> Can we do this in another pr?

I do this in https://github.com/intel-analytics/analytics-zoo/pull/4835

**Comment by Le-Zheng** [comment]
> > > @charlieJ107 I find https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> > 
> > 
> > Can we do this in another pr?
> 
> Seems these two files are same. Which one should we delete?

https://github.com/intel-analytics/analytics-zoo/pull/4835 has deleted this.

**Comment by jason-dai** [comment]
> > @charlieJ107 I find [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net?rgh-link-date=2021-09-26T03%3A54%3A33Z) is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> 
> Can we do this in another pr?

It seems that some of the scala codes are in `dllib`: https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net?

We should make it consistent for the python and scala code; For instance, should `GraphNet` be in `dllib` or `orca`?

File: python/orca/src/bigdl/orca/net/net_load.py [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+# limitations under the License.
+#
+
+import importlib
+import os
+import sys
+
+from bigdl.dllib.utils.file_utils import callZooFunc
+from bigdl.dllib.nn.layer import Model as BModel
+from bigdl.dllib.net.graph_net import GraphNet
```
Comment (yangw1234) [inline python/orca/src/bigdl/orca/net/net_load.py:23]: `bigdl.orca.net.graph_net`

**Comment by yangw1234** [comment]
> > > @charlieJ107 I find [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net?rgh-link-date=2021-09-26T03%3A54%3A33Z) is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> > 
> > 
> > Can we do this in another pr?
> 
> It seems that some of the scala codes are in `dllib`: https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net?
> 
> We should make it consistent for the python and scala code; For instance, should `GraphNet` be in `dllib` or `orca`?
@jason-dai 
How about moving `com.intel.analytics.bigdl.dllib.net` to `com.intel.analytics.bigdl.orca.net`? 

Currently the python-java interface code is implemented in orca, so the python code must go into orca.

What's our initial design/considerations for putting it in dllib.

**Comment by jason-dai** [comment]
> > > > @charlieJ107 I find [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net?rgh-link-date=2021-09-26T06%3A30%3A47Z) is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> > > 
> > > 
> > > Can we do this in another pr?
> > 
> > 
> > It seems that some of the scala codes are in `dllib`: [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net?rgh-link-date=2021-09-26T06%3A30%3A47Z)?
> > We should make it consistent for the python and scala code; For instance, should `GraphNet` be in `dllib` or `orca`?
> 
> @jason-dai
> How about moving `com.intel.analytics.bigdl.dllib.net` to `com.intel.analytics.bigdl.orca.net`?
> 
> Currently the python-java interface code is implemented in orca, so the python code must go into orca.
> 
> What's our initial design/considerations for putting it in dllib.

Why can't you put that in dllib pyhon folder?

I believe dllib.keras depends on `GraphNet` @Le-Zheng

**Comment by yangw1234** [comment]
> > > > > @charlieJ107 I find [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net?rgh-link-date=2021-09-26T06%3A30%3A47Z) is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> > > > 
> > > > 
> > > > Can we do this in another pr?
> > > 
> > > 
> > > It seems that some of the scala codes are in `dllib`: [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net?rgh-link-date=2021-09-26T06%3A30%3A47Z)?
> > > We should make it consistent for the python and scala code; For instance, should `GraphNet` be in `dllib` or `orca`?
> > 
> > 
> > @jason-dai
> > How about moving `com.intel.analytics.bigdl.dllib.net` to `com.intel.analytics.bigdl.orca.net`?
> > Currently the python-java interface code is implemented in orca, so the python code must go into orca.
> > What's our initial design/considerations for putting it in dllib.
> 
> Why can't you put that in dllib pyhon folder?
> 
> I believe dllib.keras depends on `GraphNet` @Le-Zheng

I am ok with either way, just not sure what decisions have been made on which goes where.
Let's put everything back to dllib then? @jason-dai @Le-Zheng

**Comment by jason-dai** [comment]
> > > > > > @charlieJ107 I find [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net?rgh-link-date=2021-09-26T07%3A13%3A57Z) is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> > > > > 
> > > > > 
> > > > > Can we do this in another pr?
> > > > 
> > > > 
> > > > It seems that some of the scala codes are in `dllib`: [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net?rgh-link-date=2021-09-26T07%3A13%3A57Z)?
> > > > We should make it consistent for the python and scala code; For instance, should `GraphNet` be in `dllib` or `orca`?
> > > 
> > > 
> > > @jason-dai
> > > How about moving `com.intel.analytics.bigdl.dllib.net` to `com.intel.analytics.bigdl.orca.net`?
> > > Currently the python-java interface code is implemented in orca, so the python code must go into orca.
> > > What's our initial design/considerations for putting it in dllib.
> > 
> > 
> > Why can't you put that in dllib pyhon folder?
> > I believe dllib.keras depends on `GraphNet` @Le-Zheng
> 
> I am ok with either way, just not sure what decisions have been made on which goes where.
> Let's put everything back to dllib then? @jason-dai @Le-Zheng

The basic principle is that all TensorFlow/PyTorch related functionalities are in `orca`, and BigDL releated functionalities are in `dllib`

**Comment by Le-Zheng** [comment]
> > > > > > > @charlieJ107 I find [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/python/dllib/src/bigdl/dllib/net?rgh-link-date=2021-09-26T07%3A13%3A57Z) is not used in dllib. Please help to delete it dllib. Keep migrating net in orca with the whole classes. And update some import paths "from bigdl.dllib.net.utils import .." in orca.
> > > > > > 
> > > > > > 
> > > > > > Can we do this in another pr?
> > > > > 
> > > > > 
> > > > > It seems that some of the scala codes are in `dllib`: [https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net](https://github.com/intel-analytics/analytics-zoo/tree/bigdl-2.0/scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/net?rgh-link-date=2021-09-26T07%3A13%3A57Z)?
> > > > > We should make it consistent for the python and scala code; For instance, should `GraphNet` be in `dllib` or `orca`?
> > > > 
> > > > 
> > > > @jason-dai
> > > > How about moving `com.intel.analytics.bigdl.dllib.net` to `com.intel.analytics.bigdl.orca.net`?
> > > > Currently the python-java interface code is implemented in orca, so the python code must go into orca.
> > > > What's our initial design/considerations for putting it in dllib.
> > > 
> > > 
> > > Why can't you put that in dllib pyhon folder?
> > > I believe dllib.keras depends on `GraphNet` @Le-Zheng
> > 
> > 
> > I am ok with either way, just not sure what decisions have been made on which goes where.
> > Let's put everything back to dllib then? @jason-dai @Le-Zheng
> 
> The basic principle is that all TensorFlow/PyTorch related functionalities are in `orca`, and BigDL releated functionalities are in `dllib`

Yes. dllib.keras requires `GraphNet`. Actually, we need to migrate the net folder to dllib because `net_load.py` and `graph_net.py  `  are required in dllib.

**Pull request closed without merging**